src/unitxt/metrics.py

```diff
@@ -191,24 +259,32 @@ def process(self, stream: Stream, stream_name: str = None) -> Generator:
         for instance in instances:
             yield instance
 
-    def _compute(self, references: List[str], prediction: str) -> dict:
-        result = self.compute(references=references, prediction=prediction)
-        result["score"] = result[self.main_score]
-        result["score_name"] = self.main_score
-        return result
+    @abstractmethod
+    def compute_with_additional_inputs(self, references: List[Any], prediction: Any, additional_inputs: Dict) -> dict:
+        pass
+
+
+class ReferenceBasedInstanceMetric(InstanceMetric):
+    def prepare(self):
+        super().prepare()
+        self.use_additional_inputs = False
+
+    def compute_with_additional_inputs(self, references: List[Any], prediction: Any, additional_inputs: Dict) -> dict:
+        assert additional_inputs is None
+        return self.compute(references, prediction)
 
     @abstractmethod
-    def compute(self, references: List[str], prediction: str) -> dict:
+    def compute(self, references: List[Any], prediction: Any, additional_inputs: Dict) -> dict:
```

Comment:
why pass `additional_inputs` here? isn't `compute_with_additional_inputs` for that?

Reply:
Fixed.